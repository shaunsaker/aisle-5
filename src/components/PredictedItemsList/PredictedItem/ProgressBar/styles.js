import { StyleSheet } from 'react-native';

import styleConstants from '../../../../styleConstants';

const styles = StyleSheet.create({
  wrapper: {
    flex: 1,
    alignItems: 'stretch',
    borderRadius: styleConstants.dimensions.borderRadius,
    overflow: 'hidden',
  },
  container: {
    borderRadius: styleConstants.dimensions.borderRadius,
    position: 'absolute',
    top: styleConstants.dimensions.spacing.vertical / 2,
    bottom: styleConstants.dimensions.spacing.vertical / 2,
    left: styleConstants.dimensions.spacing.horizontal / 2,
    backgroundColor: styleConstants.colors.white,
    ...styleConstants.shadows.small,
  },
});

export default styles;
