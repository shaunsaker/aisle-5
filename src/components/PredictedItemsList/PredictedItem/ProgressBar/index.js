import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';

import utils from '../../../../utils';
import styleConstants from '../../../../styleConstants';
import styles from './styles';

const propTypes = {
  value: PropTypes.number,
};

const defaultProps = {};

const ProgressBar = ({ value }) => {
  const valueAsInt = value / 100;
  const wrapperColor = utils.app.getColor(1 - valueAsInt);

  return (
    <View style={[styles.wrapper, { backgroundColor: wrapperColor }]}>
      <View style={[styles.container, { width: value }]} />
    </View>
  );
};

ProgressBar.propTypes = propTypes;
ProgressBar.defaultProps = defaultProps;

export default ProgressBar;
