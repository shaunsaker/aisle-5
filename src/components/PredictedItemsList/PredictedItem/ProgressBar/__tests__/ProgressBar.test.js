import React from 'react';
import renderer from 'react-test-renderer';

import ProgressBar from '..';

// Fixes _bezier is not a function bug
jest.useFakeTimers();

describe('ProgressBar', () => {
  describe('renders', () => {
    const value = 50;

    it('renders with minimum required props', () => {
      const component = renderer.create(<ProgressBar value={value} />);

      expect(component).toMatchSnapshot();
    });
  });
});
