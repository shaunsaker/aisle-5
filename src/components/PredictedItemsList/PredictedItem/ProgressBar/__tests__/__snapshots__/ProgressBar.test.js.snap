// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar renders renders with minimum required props 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "rgba(42, 213, 42, 1)",
      "borderRadius": 5,
      "flex": 1,
      "padding": 7.5,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#EDE8CA",
        "borderColor": null,
        "borderRadius": 5,
        "borderWidth": 0,
        "bottom": 5,
        "elevation": 2,
        "left": 7.5,
        "position": "absolute",
        "right": 7.5,
        "shadowColor": "#212121",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.17,
        "shadowRadius": 2,
        "top": 5,
        "width": "100%",
      }
    }
  />
</View>
`;
